### ISSUE TITLE
Reuse JS Player Data when multiple stream URL please ?

### ISSUE THREAD

**Issue by greentornado** [kkdai/youtube#140]
When I loop like this 

```
for _, s := range video.Formats {
	formatURL, e := yclient.GetStreamURL(video, &s)
	if e == nil {

           // It called each time here
           // 2021/01/03 11:02:36 GET https://youtube.com/embed/TVahULAGAm4?hl=en
           // 2021/01/03 11:02:37 GET https://youtube.com/s/player/5dd3f3b2/player_ias.vflset/en_US/base.js

	}
}
```

There should be some way to cache html data of embed page and base.js ....

**Comment by Julian-Chu** [comment]
hi @greentornado ,
 you just want to cache data  inside the for-loop , right?  I am considering  how and when to clean the cache.

**Comment by greentornado** [comment]
maybe by calling a clean() function ?

**Comment by Julian-Chu** [comment]
made a simple cache for single video id, of course it could be replaced by bigcache or freecache later.

**Comment by Julian-Chu** [comment]
closed by #142 
if any problem, please feel free to reopen this issue
